CMakeLists.txt

```diff
@@ -53,22 +36,68 @@ mark_as_advanced(MATERIALX_PYTHON_EXECUTABLE)
 mark_as_advanced(MATERIALX_PYTHON_OCIO_DIR)
 mark_as_advanced(MATERIALX_PYTHON_PYBIND11_DIR)
 
-mark_as_advanced(MATERIALX_BUILD_RENDER)
-mark_as_advanced(MATERIALX_TEST_RENDER)
-mark_as_advanced(MATERIALX_OSLC_EXECUTABLE)
-mark_as_advanced(MATERIALX_TESTSHADE_EXECUTABLE)
-mark_as_advanced(MATERIALX_TESTRENDER_EXECUTABLE)
-mark_as_advanced(MATERIALX_OSL_INCLUDE_PATH)
+# Shader generation options
+option(MATERIALX_BUILD_GEN_OSL "Build the OSL shader generator back-end." ON)
 mark_as_advanced(MATERIALX_BUILD_GEN_OSL)
+option(MATERIALX_BUILD_GEN_GLSL "Build the GLSL shader generator back-end." ON)
 mark_as_advanced(MATERIALX_BUILD_GEN_GLSL)
+option(MATERIALX_BUILD_GEN_OGSFX "Build the OGSFX shader generator back-end." OFF)
 mark_as_advanced(MATERIALX_BUILD_GEN_OGSFX)
+option(MATERIALX_BUILD_GEN_ARNOLD "Build the Arnold OSL shader generator back-end." OFF)
 mark_as_advanced(MATERIALX_BUILD_GEN_ARNOLD)
-mark_as_advanced(MATERIALX_BUILD_CONTRIB)
+if (NOT MATERIALX_BUILD_GEN_OSL)
+    set (MATERIALX_BUILD_GEN_ARNOLD OFF)
+endif()
+if (NOT MATERIALX_BUILD_GEN_GLSL)
+    set (MATERIALX_BUILD_GEN_OGSFX OFF)
+endif()
 
-add_definitions(-DMATERIALX_OSLC_EXECUTABLE=\"${MATERIALX_OSLC_EXECUTABLE}\")
-add_definitions(-DMATERIALX_TESTSHADE_EXECUTABLE=\"${MATERIALX_TESTSHADE_EXECUTABLE}\")
-add_definitions(-DMATERIALX_TESTRENDER_EXECUTABLE=\"${MATERIALX_TESTRENDER_EXECUTABLE}\")
-add_definitions(-DMATERIALX_OSL_INCLUDE_PATH=\"${MATERIALX_OSL_INCLUDE_PATH}\")
+# Shader render / validation options
+option(MATERIALX_BUILD_RENDER "Build the MaterialX Render module." ON)
+mark_as_advanced(MATERIALX_BUILD_RENDER)
+option(MATERIALX_BUILD_RENDEROSL "Build OSL shader generator validation." ON)
+mark_as_advanced(MATERIALX_BUILD_RENDEROSL)
+option(MATERIALX_BUILD_RENDERGLSL "Build GLSL shader generator validation." ON)
+mark_as_advanced(MATERIALX_BUILD_RENDERGLSL)
+option(MATERIALX_TEST_RENDER "Run tests for MaterialX Render module. GPU setup required for graphics validation.")
```

Comment:
Should this be marked as OFF? Doesn't seem to be ON or OFF at the moment.

Reply:
Should be off by default, but will make it explicit.